File: Frontend/src/types/task.ts
export type TaskStatus = 'To Do' | 'In Progress' | 'In Review' | 'Done';
export type TaskPriority = 'Low' | 'Medium' | 'High';

export interface Task {
  id: string;
  title: string;
  description?: string;
  status: TaskStatus;
  priority: TaskPriority;
  assignee: string;
  dueDate?: string;
  progress?: number;
  project?: string;
}

File: Backend/models/task.go
package models

import "time"

type Task struct {
	ID          uint      `gorm:"primaryKey"`
	Title       string
	Description string
	Status      TaskStatus
	Priority    TaskPriority
	DueDate     *time.Time
	ProjectID   uint
	Project     Project
	AssignedTo  *uint
	Assignee    *User     `gorm:"foreignKey:AssignedTo"`
	CreatedAt   time.Time
	UpdatedAt   time.Time

	Comments []Comment
	Labels   []TaskLabel
}
